verifiers/internal/gha/npm_sigstore_tuf_test.go

```diff
@@ -0,0 +1,187 @@
+				PublicKey: publicKey{
+					RawBytes:   "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAE1Olb3zMAFFxXKHiIkQO5cJ3Yhl5i6UPp+IhuteBJbuHcA5UogKo0EWtlWwW6KSaKoTNEYL7JlCQiVnkhBktUgg==",
+					KeyDetails: "PKIX_ECDSA_P256_SHA_256",
+					ValidFor: validFor{
+						Start: time.Date(1999, time.January, 1, 0, 0, 0, 0, time.UTC),
+					},
+				},
+			},
+			{
+				KeyID:    "SHA256:jl3bwswu80PjjokCgh0o2w5c2U4LhQAE57gj9cz1kzA",
+				KeyUsage: "npm:attestations",
+				PublicKey: publicKey{
+					RawBytes:   "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAE1Olb3zMAFFxXKHiIkQO5cJ3Yhl5i6UPp+IhuteBJbuHcA5UogKo0EWtlWwW6KSaKoTNEYL7JlCQiVnkhBktUgg==",
+					KeyDetails: "PKIX_ECDSA_P256_SHA_256",
+					ValidFor: validFor{
+						Start: time.Date(2022, time.December, 1, 0, 0, 0, 0, time.UTC),
+					},
+				},
+			},
+		},
+	}
+	testTargetKey      = testTargetKeys.Keys[1]
+	testTargetKeyID    = testTargetKey.KeyID
+	testTargetKeyUsage = testTargetKey.KeyUsage
+	testTargetKeyData  = testTargetKey.PublicKey.RawBytes
+)
+
+// mockSigstoreTufClient a mock implementation of sigstoreTufClient.
+type mockSigstoreTufClient struct {
+	fileContentMap map[string]string
+}
+
+// NewMockSigstoreTufClient returns an instance of the mock client,
+// with fileContentMap as input and outputs of the GetTarget() method.
+func NewMockSigstoreTufClient() *mockSigstoreTufClient {
+	return &mockSigstoreTufClient{fileContentMap: mockFileContentMap}
+}
+
+// GetTarget mock implementation of GetTarget for the mockSigstoreTufClient.
+func (c mockSigstoreTufClient) GetTarget(targetPath string) ([]byte, error) {
+	content, exists := c.fileContentMap[targetPath]
+	if !exists {
+		return nil, fmt.Errorf("content not definied in this mock, key: %s", targetPath)
+	}
+	return []byte(content), nil
+}
+
+// TestGetNpmjsKeysTarget ensures we can parse the target file.
+func TestGetNpmjsKeysTarget(t *testing.T) {
+	tests := []struct {
+		name         string
+		targetPath   string
+		expectedKeys *npmjsKeysTarget
+		expectedErr  error
+	}{
+		{
+			name:         "parsing local registry.npmjs.org_keys.json",
+			targetPath:   testTargetPath,
+			expectedKeys: &testTargetKeys,
+		},
+		{
+			name:        "parsing non-existent registry.npmjs.org_keys.json",
+			targetPath:  "my-fake-path.json",
+			expectedErr: ErrorCouldNotFindTarget,
+		},
+		{
+			name:        "parsing invalid json",
+			targetPath:  testTargetInvalidJSONPath,
+			expectedErr: ErrorCouldNotParseKeys,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mockClient := NewMockSigstoreTufClient()
+			actualKeys, err := getNpmjsKeysTarget(mockClient, tt.targetPath)
+			if keyDataDiff := cmp.Diff(tt.expectedKeys, actualKeys, cmpopts.EquateComparable()); keyDataDiff != "" {
+				t.Errorf("expected equal values (-want +got):\n%s", keyDataDiff)
+			}
+			if errorDiff := cmp.Diff(tt.expectedErr, err, cmpopts.EquateErrors()); errorDiff != "" {
+				log.Printf("want: %v, got: %v, type:%T", tt.expectedErr, err, err)
```

Comment:
nit: Do you need this message if you call `t.Error` right after it?
```suggestion
```

Reply:
I was debugging something again, after the long pause. I'll convert this PR back to a draft for now, while we decide on how exactly to get the KeyID dynamically.